internal/resource/config/serversettings/server_settings_system_keys_resource.go

```diff
@@ -0,0 +1,396 @@
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &serverSettingsSystemKeysResource{}
+	_ resource.ResourceWithConfigure   = &serverSettingsSystemKeysResource{}
+	_ resource.ResourceWithImportState = &serverSettingsSystemKeysResource{}
+)
+
+// ServerSettingsSystemKeysResource is a helper function to simplify the provider implementation.
+func ServerSettingsSystemKeysResource() resource.Resource {
+	return &serverSettingsSystemKeysResource{}
+}
+
+// serverSettingsSystemKeysResource is the resource implementation.
+type serverSettingsSystemKeysResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type serverSettingsSystemKeysResourceModel struct {
+	Id       types.String `tfsdk:"id"`
+	Current  types.Object `tfsdk:"current"`
+	Previous types.Object `tfsdk:"previous"`
+	Pending  types.Object `tfsdk:"pending"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *serverSettingsSystemKeysResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	serverSettingsSystemKeysResourceSchema(ctx, req, resp, false)
+}
+
+func serverSettingsSystemKeysResourceSchema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse, setOptionalToComputed bool) {
+	schema := schema.Schema{
+		Description: "Manages a Server Settings SystemKeys.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Placeholder for Terraform",
+				Computed:    true,
+				Optional:    false,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"current": schema.SingleNestedAttribute{
+				Description: "Current SystemKeys Secrets that are used in cryptographic operations to generate and consume internal tokens.",
+				Required:    true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"creation_date": schema.StringAttribute{
+						Description: "Creation time of the key.",
+						Computed:    true,
+						Optional:    false,
+						Required:    false,
+					},
+					"encrypted_key_data": schema.StringAttribute{
+						Description: "The system key encrypted.",
+						Computed:    true,
+						Optional:    true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+					"key_data": schema.StringAttribute{
+						Description: "The clear text system key base 64 encoded. The system key must be 32 bytes before base 64 encoding",
+						Computed:    true,
+						Optional:    false,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
```

Comment:
If this is computed fully by PF and we don't need to pass it to the API, then we shouldn't need UseStateForUnknown, same for the other key_data fields